### ISSUE TITLE
bolt.new cannot open the private Github repository: no access to the repository

### ISSUE THREAD

**Issue by 1WorldCapture** [stackblitz/bolt.new#95]
### Describe the bug

A Github private repository opened in stackblitz cannot be opened and loaded with bolt.new. When loading this repository using bolt.new, bolt.new reports an error as follows: ![image](https://github.com/user-attachments/assets/6530c3df-e65b-4581-9560-9a874d37a7cf)

### Link to the Bolt URL that caused the error

https://bolt.new/~/github.com/1WorldCapture/ship-fast-ts-supabase

### Steps to reproduce

1. open the Github private repository in Stackblitz; 
2. get the project ID via browser url, e.g. `https://stackblitz.com/~/github.com/1WorldCapture/ship-fast-ts-supabase?file=config.ts`, the project ID is `/github.com/1WorldCapture/ship-fast-ts-supabase`; 
3. Construct the bolt.new address (https://bolt.new/~/ <project id="">and open it with your browser; 
4. bolt.new initializes and imports with the error "You don't have access to the repository"

### Expected behavior

Expect bolt.new to load private repository projects in stackblitz correctly, and to be able to modify the project code through bolt.new's dialogs properly

### Platform

- OS: Windows
- Browser: Chrome
- Version: 129.0.6668.90

**Comment by tiagoefreitas** [comment]
I also need this to edit my existing projects or to start from a template

**Comment by 1WorldCapture** [comment]
> I also need this to edit my existing projects or to start from a template

Agree. If we need to build something real, it must start from some base (existing project or own project template), not blank template......

**Comment by kirjavascript** [comment]
Hi! 👋 Thanks a lot for the report!

We're tracking this issue internally and will let you know when we have a resolution.

**Comment by danielpox** [comment]
I created a project using Bolt, and then I opened it in Stackblitz and from there I created a Github repository. I made it private, and I can make changes in the Stackblitz editor, but the connection to Bolt is lost. And as mentioned in this issue, if I try to open the repo in Bolt using the correct link structure, I get the import failed message.

**Comment by 1WorldCapture** [comment]
> Hi! 👋 Thanks a lot for the report!
> 
> We're tracking this issue internally and will let you know when we have a resolution.

Glad to hear this issue is on track, and pls. help speed this problem solving up, it's really important for bolt.new to be production tool

**Comment by lipecss** [comment]
> I created a project using Bolt, and then I opened it in Stackblitz and from there I created a Github repository. I made it private, and I can make changes in the Stackblitz editor, but the connection to Bolt is lost. And as mentioned in this issue, if I try to open the repo in Bolt using the correct link structure, I get the import failed message.
> 
> Also I thought upgrading to the paid Pro plan would solve the issue, but those $9 did not help. 😅

I also noticed this problem

**Comment by 136sergio** [comment]
Bolt is still not able to open private github repositories, you must change your repository to public and it will open-

**Comment by Jonkitonki** [comment]
Did anyone archieve some progress on this issue? It would be great to keep some projects on private in github :-)

**Comment by Veeeetzzzz** [comment]
> Did anyone archieve some progress on this issue? It would be great to keep some projects on private in github :-)

![image](https://github.com/user-attachments/assets/0db201b4-4b3e-41d7-be78-ee216bb66685)

Yeah +1 - I can use public repos and I'd happily upgrade to a Teams plan if I was working with organisational information that needs to stay private. Right now I'm just making the repos I need to import into Bolt.new public and then reverting or creating a public fork to work off.

Will be so good when this is implemented.

**Comment by Nic-Tai** [comment]
i still face this issue. may i know what's the solution?

**Comment by kirjavascript** [comment]
> Is there any workaround for this problem for now? @kirjavascript

I'm afraid the only workaround for now is copying the bits you need for bolt into a public repo.

**Comment by Greyvend** [comment]
@kirjavascript any ETA on when this might be fixed?

**Comment by codetesta2z** [comment]
This is really defeating the overall purpose of using bolt. Can we expedite it please.

**Comment by fvrlak** [comment]
It has been almost two months, and the issue still hasn't been solved.

I don't understand the thinking of bolt.new team. Do they really think that importing private repo from github is not essential?

What am I supposed to do with the project after being developed? Put it as public and then copy a link and make it private again?

It is such an important thing even essential thing to have an option to get repo imported back to bolt, that now currently I am not even using the app simply because there is no point in creating any new application.

If it is for some reason not possible to make it via github, then at least it could be possible to upload the folder from pc.

**Comment by jannikMetzner** [comment]
> It has been almost two months, and the issue still hasn't been solved.
> 
> I don't understand the thinking of bolt.new team. Do they really think that importing private repo from github is not essential?
> 
> What am I supposed to do with the project after being developed? Put it as public and then copy a link and make it private again?
> 
> It is such an important thing even essential thing to have an option to get repo imported back to bolt, that now currently I am not even using the app simply because there is no point in creating any new application.
> 
> If it is for some reason not possible to make it via github, then at least it could be possible to upload the folder from pc.

same here, have to stop using bolt.new, because I can't sustain my dev process without tracking changes in git

**Comment by Tyberious** [comment]
Not sure why this wasn't and still isn't implemented out of the gate, not having it defeats the majority of its purpose.

**Comment by Greyvend** [comment]
> You could you use the `bolt to github` extension in chrome and follow this tutorial [youtube.com/watch?v=d9NqXRoroi0](https://www.youtube.com/watch?v=d9NqXRoroi0)

It temporarily creates a public copy of your private repo, which is not secure. It's possible just to make your private repo public, then import, then switch it back, but that's an unacceptable workaround for many.

Besides, there needs to be a way to sync changes between the GitHub repo and the Bolt project continuously.

**Comment by eksno** [comment]
Big issue for me too. Got a team plan for this purpose just to realize it doesn't support private repositories...

**Comment by avelops** [comment]
Any news on this issue?

**Comment by champcbg** [comment]
This is a real blocker any updates on this.  It is go on 8 months of a known issue, but also huge blocker.  At least us go away if there is no plan to fix it.

**Comment by Olalexy1** [comment]
https://support.bolt.new/troubleshooting/integrations-issues#github-authentication-issues I applied the github authentication fix and it resolved my problem.
